ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -399,10 +399,14 @@ let pool_migrate ~__context ~vm ~host ~options =
     use_compression ~__context options (Helpers.get_localhost ~__context) host
   in
   debug "%s using stream compression=%b" __FUNCTION__ compress ;
-  let ip = Http.Url.maybe_wrap_IPv6_literal address in
-  let scheme = if !Xapi_globs.migration_https_only then "https" else "http" in
+  let http = if !Xapi_globs.migration_https_only then "https" else "http" in
```

Comment:
This should be `scheme`, because it can be either

Reply:
But `scheme` inside `URI` is bound and shadows it when using it in `URI.(make ... scheme)`. So I would have to give up opening `URI` to use `scheme`.

Reply:
Wouldn't that make it simpler?

    Uri.(
          make ~scheme ~host:...

Reply:
The problem with `Uri.(..)` is that `scheme` and so on are functions inside `Uri` that shadow my bindings. So now I prefer to not open `Uri` and avoid this.

Reply:
Oh I see... In the following commit you have `Uri.make ... |> Uri.to_string` to avoid this.